llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -305,6 +310,15 @@ static bool hasAssertInFunctionCallGraph(llvm::Function *Func) {
     if (F != Func)
       FuncCallStack.push_back(F);
 
+    bool HasIndirectlyCalledAttr = false;
+    if (F->hasFnAttribute("referenced-indirectly") ||
+        std::find(isIndirectlyCalledInGraph.begin(),
+                  isIndirectlyCalledInGraph.end(),
+                  F) != isIndirectlyCalledInGraph.end()) {
+      HasIndirectlyCalledAttr = true;
+      isIndirectlyCalledInGraph.push_back(F);
```

Comment:
If F has been found in isIndirectlyCalledInGraph then it will be included there twice.

Reply:
fixed in 69077b9 and b576223